docs: fix links to config-files documents

glyph-se opened the pull request:
While working on https://github.com/zwave-js/node-zwave-js/pull/2854 I checked the CONTRIBUTING.md and saw some invalid links, just helping out and fixing them :)

AlCalzone commented:
Thanks - do you think it would make more sense to link to the documentation instead of its source files?
=> https://zwave-js.github.io/node-zwave-js/#/config-files/overview etc.

glyph-se commented:
> 
> 
> Thanks - do you think it would make more sense to link to the documentation instead of its source files?
> => https://zwave-js.github.io/node-zwave-js/#/config-files/overview etc.

Yes, I think this makes. The online documentation is much easier to navigate largely due to the left menu.

I changed the PR slightly, now pointing just to the general instructions for device configuration files, less duplication of content.

I also moved the style guide description further to the top on that page as I believe that to be more important and general than the others. To me partial parameters and conditional parameters seems more like edge cases that are not always relevant, while the style guide always is important.

What do you think?

AlCalzone commented:
Sounds good!
